### ISSUE TITLE
Hard Reload cache bypass for whitelisted IP addresses

### ISSUE THREAD

**Issue by AntonEvers** [fastly/fastly-magento2#147]
I've worked with a Magento 1 varnish plugin that provided the option to bypass cache on a hard reload from the browser (`CMD + SHIFT + R` or `SHIFT + F5`).

Only for a whitelisted set of IP addresses this gave the shop owner the opportunity to peek behind the cache to see if changes were already present in Magento and would appear after a varnish flush.

Since this is probably not such a big change, would you be open to adding it to your plugin as well?

**Comment by vvuksan** [comment]
I am unclear how that is achieved. Does the browser indicate somehow that it's a hard reload ?

**Comment by ajpevers** [comment]
Yes indeed. A hard refresh has the request headers:
```
pragma: no-cache
cache-control: no-cache
```

Whereas a normal refresh has the headers:
```
cache-control: max-age=0
```

I compared the requests for google.com in my network tab in chrome:

Normal refresh:
```
curl 'https://www.google.nl/' -H 'accept-encoding: gzip, deflate, br' -H 'accept-language: en-US,en;q=0.9,nl;q=0.8,fr;q=0.7' -H 'upgrade-insecure-requests: 1' -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.186 Safari/537.36' -H 'x-chrome-uma-enabled: 1' -H 'accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8' -H 'cache-control: max-age=0' -H 'authority: www.google.nl' -H 'cookie: SID=0wWAMPIA6S8liB23XqbetMITmI-FbFqwkdV5cR0SOznyctVimh1liSnyx-uNvCHoVDGeeA.; HSID=Akhb9QuBUzFgJz1p4; SSID=AmqpM25cdRHRE9r2c; APISID=kCVg8BYIKoQ0mObF/AfCTkvkxGcPZwTC4q; SAPISID=sD8ESFIAMGV-CmL6/AJKSTmKNE2C6T_3O4; DV=c7FZtRIEYZsQ8E2ae6M2hb1L4i88IBY; NID=125=cYolIN1TqvnGikgUY7pGBUs2_jmExdJ9Yx6q_V7F-IvjJPEuUWbtGbSCwVI_dsUHw_tewPojzvp9ASmPEkY1Fw1jK0tahF7zAU5zR64evQjxvIhGP7xiddUKhmVbXEolx6_9RxCDDIGhkN6u53wGu9yXvZmqy_fyhGMTVSZhNZPypfyT6t7dJnDeAnP2fQSxXW1MQ6v90TK-t8aUb87Tl8qt5jJViB5gRGsZL2EtXTN-3Tzt1K1hZLvmoJkcZ44i7J3DRVX7Ni2cFskfHSJKGi2AQ4V95h6wbazRQxsgtlhcsaXaKvvQkE1Ia8Cxgv5_4mnuVvxAbViufG4kc5r3OUabv9zD5x-OE8qMvA; 1P_JAR=2018-3-8-3; UULE=a+cm9sZToxIHByb2R1Y2VyOjEyIHByb3ZlbmFuY2U6NiB0aW1lc3RhbXA6MTUyMDQ4MTU3MDkzMjAwMCBsYXRsbmd7bGF0aXR1ZGVfZTc6MjM3OTAxNzY4IGxvbmdpdHVkZV9lNzo5MDQwMjI4NzB9IHJhZGl1czoxMjQwMA==' -H 'x-client-data: CJO2yQEIpLbJAQjEtskBCKmdygEIqKPKAQ==' --compressed
```

Hard refresh
```
curl 'https://www.google.nl/' -H 'pragma: no-cache' -H 'accept-encoding: gzip, deflate, br' -H 'accept-language: en-US,en;q=0.9,nl;q=0.8,fr;q=0.7' -H 'upgrade-insecure-requests: 1' -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.186 Safari/537.36' -H 'x-chrome-uma-enabled: 1' -H 'accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8' -H 'cache-control: no-cache' -H 'authority: www.google.nl' -H 'cookie: SID=0wWAMPIA6S8liB23XqbetMITmI-FbFqwkdV5cR0SOznyctVimh1liSnyx-uNvCHoVDGeeA.; HSID=Akhb9QuBUzFgJz1p4; SSID=AmqpM25cdRHRE9r2c; APISID=kCVg8BYIKoQ0mObF/AfCTkvkxGcPZwTC4q; SAPISID=sD8ESFIAMGV-CmL6/AJKSTmKNE2C6T_3O4; DV=c7FZtRIEYZsQ8E2ae6M2hb1L4i88IBY; NID=125=AAKancpHUQJfw-W6GKxAwBiZEPyE2YOb8xeZgdNrAzFUJFNj9MUGAmqEAP-ptSu6zc9m0bBHhrJlKw-EgGqnJbbpZfWHe4eirT3AGG3JTyZ54TPNaMjmy9yM3itvBimV8F-II-FsAKMYkJqV69wjWylaTAfNUWVBqaTDAc0z-ooeC-sx24AtOe7jbA1gS3EIdHFb9mTwKkIFXGxniP2fDuUNVgCyp-4eWffgX1bw03IJ1BZJ8d1jU7ycxteXoMVsWPeCu23X47YUox5xgqkYVOWu2PiU2QIV-fQHF-efNmbqmdWmSFNb_bDRhFMthRUt9qgnHf1iIP6OcH-J68ndygnFJlxTzcjVImE3kg; UULE=a+cm9sZToxIHByb2R1Y2VyOjEyIHByb3ZlbmFuY2U6NiB0aW1lc3RhbXA6MTUyMDQ4MTU4NDU2NDAwMCBsYXRsbmd7bGF0aXR1ZGVfZTc6MjM3OTAxODA0IGxvbmdpdHVkZV9lNzo5MDQwMjI4NTh9IHJhZGl1czoxMjQwMA==; 1P_JAR=2018-3-8-3' -H 'x-client-data: CJO2yQEIpLbJAQjEtskBCKmdygEIqKPKAQ==' --compressed
```

**Comment by vvuksan** [comment]
We'll add this as a feature request

**Comment by vvuksan** [comment]
This has been implemented using Fastly Edge Modules. It will be available in version 1.2.90. In order to use it you will need to enable Fastly Edge Modules in Advanced config then configure the module.
